examples/common/convert_c4.py

```diff
@@ -275,15 +392,17 @@ def generate_samples(
 
 
 def main(args: Namespace) -> None:
-    """Main: create C4 streaming dataset.
+    """Main: create C4/pile streaming dataset.
 
     Args:
         args (Namespace): Commandline arguments.
     """
-    hf_splits = ('train', 'train', 'validation', 'validation')
-    folder_splits = ('train', 'train_small', 'val', 'val_small')
-    expected_counts = (364868892, 1000000, 364608, 10000)
-    truncate_counts = (None, 100000, None, 10000)
+    try:
+        dataset_constants = CONSTS[args.dataset]
+        iterate_over = dataset_constants
+    except KeyError:
+        warnings.warn(f'Constants for dataset {args.dataset} not found. Will be unable to provide execution time estimates.')
```

Comment:
are we confident it works for all hf datasets? or should this actually error instead of warn?

Reply:
Agreed with all the above. I think my worry is that for datasets we haven't tried (e.g. not the pile or c4), this script actually doesn't work for some reason that we haven't discovered yet. So was just wondering how confident we were that it works for all hf datasets.

Reply:
I think we're not confident. Should we have it error out if a dataset is selected that's not in `CONSTANTS`?

Reply:
I think lets error out if its not pile/c4 for now. We can easily add/test more, and a user can easily remove that error and use whatever they want at their own risk. Maybe have the error say something like "we've only tests on c4/pile. it may extend to other hf datasets. please open an issue if one interests you!"